openassessment/xblock/ui_mixins/mfe/ora_config_serializer.py

```diff
@@ -156,42 +145,60 @@ def _get_step(self, rubric_assessments, step_name):
                 return step
         return None
 
-    def __init__(self, *args, **kwargs):
-        self.step_name = kwargs.pop("step_name")
-        super().__init__(*args, **kwargs)
-
-    def to_representation(self, instance):
-        assessment_step = self._get_step(instance, self.step_name)
-
-        # Special handling for the peer step which includes extra fields
-        if assessment_step and self.step_name == "peer-assessment":
-            return PeerSettingsSerializer(assessment_step).data
-        elif assessment_step and self.step_name == "self-assessment":
-            return SelfSettingsSerializer(assessment_step).data
-        elif assessment_step and self.step_name == "student-training":
-            return StudentTrainingSettingsSerializer(assessment_step).data
-
+    # pylint: disable=arguments-renamed
+    def to_representation(self, xblock):
+        assessment_step = self._get_step(xblock.rubric_assessments, self.STEP_NAME)
         # If we didn't find a step, it is not required
         if assessment_step is None:
-            assessment_step = {"required": False}
+            return {"required": False}
+
+        assessment_step = dict(assessment_step)
+        # Add overridden start and due dates for peer assessment and self assessment
+        if self.STEP_NAME in ('peer-assessment', 'self-assessment'):
+            _, _, start, due = xblock.is_closed(step=self.STEP_NAME)
+            assessment_step['start'] = start.isoformat()
+            assessment_step['due'] = due.isoformat()
 
         return super().to_representation(assessment_step)
 
 
-class AssessmentStepsSettingsSerializer(Serializer):
-    studentTraining = AssessmentStepSettingsSerializer(
-        step_name="student-training", source="rubric_assessments"
-    )
-    peer = AssessmentStepSettingsSerializer(
-        step_name="peer-assessment", source="rubric_assessments"
+class SelfSettingsSerializer(AssessmentStepSettingsSerializer):
+    STEP_NAME = 'self-assessment'
+    startDatetime = DateTimeField(source='start')
+    endDatetime = DateTimeField(source='due')
+
+
+class StudentTrainingSettingsSerializer(AssessmentStepSettingsSerializer):
+    STEP_NAME = 'student-training'
+    numberOfExamples = SerializerMethodField(source="*", default=0)
+
+    def get_numberOfExamples(self, assessment):
+        return len(assessment["examples"])
+
+
+class PeerSettingsSerializer(AssessmentStepSettingsSerializer):
+    STEP_NAME = 'peer-assessment'
+    minNumberToGrade = IntegerField(source="must_grade")
+    minNumberToBeGradedBy = IntegerField(source="must_be_graded_by")
+
+    startDatetime = DateTimeField(source='start')
+    endDatetime = DateTimeField(source='due')
+
+    enableFlexibleGrading = BooleanField(
+        source="enable_flexible_grading", required=False
     )
+
+
+class StaffSettingsSerializer(AssessmentStepSettingsSerializer):
+    STEP_NAME = 'staff-assessment'
+
+
+class AssessmentStepsSettingsSerializer(Serializer):
```

Comment:
This diff is messy. Here's what I did.

- Changed source to "*" because we needed the fields on the block
- Removed the step name var and just baked that into the class itself, and therefore called each specific settings step class.
- Made all of the settings classes inherit from the abstract assessment step settings class, which just gives them the to_representation and the is_required field, and rather than calling the serializers from the to_representation method, we just get that behavior because they are inheriting.
- Check for the two steps with the due dates we care about, and resolve and insert the correct dates
-